contracts/interfaces/IERC1404.sol

```diff
@@ -0,0 +1,34 @@
+//SPDX-License-Identifier: MPL-2.0
+
+pragma solidity ^0.8.17;
+
+
+interface IERC1404 {
+    /**
+     * @dev See ERC-1404
+     *
+     */
+    function detectTransferRestriction(
+        address _from,
+        address _to,
+        uint256 _amount
+    ) external view returns (uint8);
+
+    /**
+     * @dev See ERC-1404
+     *
+     */
+    function messageForTransferRestriction(uint8 _restrictionCode)
+        external
+        view
+        returns (string memory);
+
+    /**
+     * @dev Returns true if the transfer is valid, and false otherwise.
+     */
+    function isTransferValid(
```

Comment:
EIP1404 (https://medium.com/erc1404/erc-1404-simple-restricted-token-standard-f71290a48faa) does not contain isTransferValid, I think we should move this function outside IERC1404.sol